tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -0,0 +1,197 @@
+// <copyright file="StatsAggregator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+using Datadog.Trace.PlatformHelpers;
+
+namespace Datadog.Trace.Agent
+{
+    internal class StatsAggregator : IStatsAggregator
+    {
+        private const int BufferCount = 2;
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<StatsAggregator>();
+
+        private readonly StatsBuffer[] _buffers;
+
+        private readonly IApi _api;
+
+        private readonly TaskCompletionSource<bool> _processExit;
+
+        private readonly TimeSpan _duration;
+
+        private readonly Task _flushTask;
+
+        private int _currentBuffer;
+
+        internal StatsAggregator(IApi api, ImmutableTracerSettings settings, TimeSpan duration)
+        {
+            _api = api;
+            _processExit = new TaskCompletionSource<bool>();
+            _duration = duration;
+            _buffers = new StatsBuffer[BufferCount];
+
+            var header = new ClientStatsPayload
+            {
+                Environment = settings.Environment,
+                ServiceName = settings.ServiceName,
```

Comment:
What is `ClientStatsPayload.ServiceName` used for? The value from settings is not always the one used by the tracer. For example, if users don't set the service name explicitly, we have fallbacks to determine a service name.

Reply:
Weird, I don't see the servicename being set in the Python implementation. We'll need to check how this is used because each aggregation key already includes a service name, so it's not clear what purpose this higher-level one has

Reply:
I also don't see the agent populating the ClientStatsPaylod with a service, so I wonder if we can omit that: https://github.com/DataDog/datadog-agent/blob/f6fb1b9388c6829cfeb18467c79f0752336e829d/pkg/trace/stats/client_stats_aggregator.go#L250-L256

Reply:
Removed the property `ClientStatsPayload.ServiceName` because it doesn't seem to be used and only adds confusion. If this is incorrect, we can adjust